### ISSUE TITLE
`update` doesn't work with a hash-map argument

### ISSUE THREAD

**Issue by pharcosyle** [lokke-org/lokke#17]
Work:
```
(update (vector 1 2) 1 inc)
;; => [2 2]
```
```
(update-in (hash-map :a 1) (vector :a) inc)
;; => {:a 2}
```

Doesn't work:
```
(update (hash-map :a 1) :a inc)
;; No applicable method for #<<generic> update (2)> in call (update #<<hash-map> 7fa3d5480f60 #<fash ((#:a . 1))>> #:a #<procedure #{1+}# (_)>)
```

**Comment by rlbdv** [comment]
Pushed the change.

Thanks again.

**Comment by pharcosyle** [comment]
Thank _you_ @rlbdv. I've been using `lokke` for about a week now and I've decided I love it and I'll be using it extensively.

I have found a number of other omissions / missed edge cases / "bugs": is it worth reporting them all as issues? For my part all are easy enough to work around. Understanding better now the scope of what you're trying to achieve here I'm wondering if you're interested in ironing out such details at this time.
